src/Data/StrMap/ST.js

```diff
@@ -7,10 +7,14 @@ exports["new"] = function () {
   return {};
 };
 
-exports.peek = function (m) {
-  return function (k) {
-    return function () {
-      return m[k];
+exports.peekImpl = function (just) {
+  return function (nothing) {
+    return function (m) {
+      return function (k) {
+        return function () {
+          return m.hasOwnProperty(k) ? just(m[k]) : nothing;
```

Comment:
`{}.hasOwnProperty.call(m, k)` to protect yourself in the case the map contains a `hasOwnProperty` key. There's a better solution, though and I will send a PR later. That would be good enough for now.